virttest/vt_resmgr/resources/pool.py

```
@@ -0,0 +1,315 @@
+        return resource.uuid
+
+    def destroy_resource_object(self, resource_id):
+        """
+        Destroy the resource object.
+        The resource object can be destroyed only when the resource is
+        released and unbound from all of its backing objects.
+        """
+        resource = self.resources.pop(resource_id)
+        LOG.debug(f"Destroy the resource object of {resource.name} in pool {self.name}")
+        if resource.allocated:
+            raise Exception("Cannot destroy an allocated resource object")
```

Ruff (0.13.1)

Create your own exception

(TRY002)